h2o-core/src/main/java/water/rapids/PermutationVarImp.java

```diff
@@ -0,0 +1,271 @@
+
+import hex.*;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.TwoDimTable;
+import water.util.VecUtils;
+
+import java.util.*;
+
+/**
+ * Permutation Variable (feature) importance measures the increase in the prediction error of the model after we permuted 
+ * the variables's values, which breaks the relationship between the variables and the true outcome.
+ * https://christophm.github.io/interpretable-ml-book/feature-importance.html
+ *
+ * Calculate permutation variables importance, by shuffling randomly each variables of the passed model
+ * then scoring the model with the newly created frame using One At a Time approach
+ * and Morris method; creating TwoDimTable with relative, scaled, and percentage value
+ *                             TwoDimTable with mean of the absolute value, and standard deviation of all features importance
+ *                             
+ * */
+
+public class PermutationVarImp {
+
+    public double[] _pVarImp; // permutation variables (feature) importance, relative value      
+    
+    private String _responseCol; 
+    
+    private String[] _features; 
+    private String[] _featuresWithNoResCol; // features without response col
+    public String[] _metrics;
+
+    /**
+     * Stores the metric to be used (loss function) original values and j-th variables's score metric values  
+     * */
+    public class LocalMetric{ // do it as a string to get the metric from the user
+
+        LocalMetric(String metric) {    m_selectedMetric = metric;  }
+        LocalMetric() {    m_selectedMetric = "mse";  } 
+        String m_selectedMetric;
+        double m_ogMetric;
+        double m_variablesMetric;
+    }
+
+    public LocalMetric _variImpMetric;
+    TwoDimTable _permutationVarImp;
+    private Model _model;
+    private Frame _trainFr; // original dataset used to train model
+
+    /**
+     * When passing only the Model without Frame (will use models frame)
+     * */
+    public PermutationVarImp(Model model) {
+        _model = model;
+        _trainFr = _model._parms._train.get();
+        init();
+    }
+
+    public PermutationVarImp(Model model, Frame data_set) {
+        _model = model;
+        _trainFr = data_set;
+        init();
+    }
+    
+    void init(){
+        _responseCol = _model._parms._response_column;
+        _features = _trainFr.names();
+
+        // regression: mse, R2, rmse, rmsele , mae
+        // classification gini, absolute_mcc, logloss, AUC, AUC_PR 
+        _metrics = new String[]{"r2", "mse", "rmse", "gini coefficient", "f1", "logloss", "auc"};
+    }
+    public String [] get_features_wo_res_test(){return _featuresWithNoResCol;}
+    
+    /**
+     * Creates a new array of Strings without the response column
+     * */
+    public void removeResCol(){
+        // remove response col from array of string to be added to TwoDimTable
+        _featuresWithNoResCol = _trainFr.names();
+        if (_featuresWithNoResCol[0].equals(_responseCol)) {
```

Comment:
The response column is not always at index 0.

Reply:
I left it for the cases which is. I have a different aproach now and am also removing the ignored variables from the params